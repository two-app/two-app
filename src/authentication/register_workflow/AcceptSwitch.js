import {StyleSheet, Switch, Text, View} from "react-native";
import PropTypes from "prop-types";
import Colors from "../../Colors";
import React, {useState} from "react";

const AcceptBox = ({children, onEmit}) => {
    const [accepted, setAccepted] = useState(false);
    return (
        <View style={styles.container}>
            <Text style={[styles.condition, accepted ? styles.acceptedCondition : undefined]}>{children}</Text>
            <View style={styles.switchContainer}>
                <Switch style={styles.switch}
                        value={accepted}
                        onValueChange={setAccepted}
                />
            </View>
        </View>
    );
};

AcceptBox.propTypes = {
    children: PropTypes.any,
    onEmit: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        flex: 1,
        justifyContent: "space-between",
        alignItems: "center",
        backgroundColor: Colors.LIGHT,
        borderRadius: 10,
        padding: 20,
        marginTop: 20
    },
    condition: {
        color: Colors.DARK,
        width: "80%"
    },
    acceptedCondition: {
        color: Colors.FADED
    },
    switchContainer: {
        width: "20%",
        flex: 1,
        alignItems: "center",
        justifyContent: "center"
    },
    switch: {
        transform: [{scaleX: .8}, {scaleY: .8}]
    }
});

export default AcceptBox;