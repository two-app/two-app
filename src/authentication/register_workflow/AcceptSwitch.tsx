import {StyleSheet, Switch, Text, View} from 'react-native';
import PropTypes from 'prop-types';
import React, {useState} from 'react';

import Colors from '../../Colors';

type AcceptBoxProps = {
  children?: any;
  onEmit: any;
  required?: boolean;
  accessibilityHint: string;
};

const AcceptBox = ({
  children,
  onEmit,
  required,
  accessibilityHint,
}: AcceptBoxProps) => {
  const [accepted, setAccepted] = useState(false);

  return (
    <View
      style={[
        styles.container,
        required ? styles.required : undefined,
        accepted ? styles.accepted : undefined,
      ]}
      data-testid="container">
      <Text
        style={[styles.condition, accepted ? styles.acceptedText : undefined]}>
        {children}
      </Text>
      <View style={styles.switchContainer}>
        <Switch
          style={styles.switch}
          value={accepted}
          onValueChange={v => {
            setAccepted(v);
            onEmit(v);
          }}
          accessibilityHint={accessibilityHint}
        />
      </View>
    </View>
  );
};

AcceptBox.propTypes = {
  onEmit: PropTypes.func.isRequired,
  isRequired: PropTypes.bool,
};

AcceptBox.defaultProps = {
  isRequired: false,
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
    borderRadius: 10,
    borderColor: Colors.LIGHT,
    borderWidth: 1,
    padding: 20,
    marginTop: 20,
  },
  required: {
    borderColor: Colors.DARK,
  },
  accepted: {
    borderColor: Colors.VALID_GREEN_DARK,
    backgroundColor: Colors.VALID_GREEN,
  },
  acceptedText: {
    color: 'white',
  },
  condition: {
    color: Colors.DARK,
    width: '80%',
  },
  switchContainer: {
    width: '20%',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  switch: {
    transform: [{scaleX: 0.8}, {scaleY: 0.8}],
  },
});

export default AcceptBox;
