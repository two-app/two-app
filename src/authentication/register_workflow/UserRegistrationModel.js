// @flow

import EmailValidator from "../../forms/EmailValidator";

/**
 * Blank UserRegistration object with Flow types.
 * @type {{firstName: null, lastName: null, password: null, email: null}}
 */
const UserRegistration: {
    firstName: string,
    lastName: string,
    email: string,
    password: string,
    acceptedTerms: boolean,
    ofAge: boolean,
    receivesEmails: boolean
} = {
    firstName: null,
    lastName: null,
    email: null,
    password: null,
    acceptedTerms: false,
    ofAge: false,
    receivesEmails: false
};

const isFirstNameValid = firstName => firstName != null && firstName.length > 1;
const isLastNameValid = lastName => lastName != null && lastName.length > 1;
const isEmailValid = email => email != null && EmailValidator.validateEmail(email);
const isPasswordValid = password => password != null && password.length > 3;

/**
 * @param ur {UserRegistration} to validate
 * @returns {boolean} true if the user supplied information is valid. Does not check terms & age.
 */
const isUserRegistrationValid = (ur: UserRegistration) => (
    isFirstNameValid(ur.firstName) &&
    isLastNameValid(ur.lastName) &&
    isEmailValid(ur.email) &&
    isPasswordValid(ur.password)
);

export default {isFirstNameValid, isLastNameValid, isEmailValid, isPasswordValid, isUserRegistrationValid};
export {UserRegistration};