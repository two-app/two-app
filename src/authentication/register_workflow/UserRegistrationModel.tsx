import "react-native-get-random-values";
import uuid from "uuidv4";

import EmailValidator from "../../forms/EmailValidator";

class UserRegistration {
  uid = uuid();
  firstName = "";
  lastName = "";
  email = "";
  password = "";
  acceptedTerms = false;
  ofAge = false;
  receivesEmails = false;
}

const isFirstNameValid = (firstName: string) =>
  firstName !== "" && firstName.length > 1;
const isLastNameValid = (lastName: string) =>
  lastName !== "" && lastName.length > 1;
const isEmailValid = (email: string) =>
  email !== "" && EmailValidator.validateEmail(email);
const isPasswordValid = (password: string) =>
  password !== "" && password.length >= 6;

const isUserRegistrationValid = (ur: UserRegistration): boolean =>
  isFirstNameValid(ur.firstName) &&
  isLastNameValid(ur.lastName) &&
  isEmailValid(ur.email) &&
  isPasswordValid(ur.password);

export default {
  isFirstNameValid,
  isLastNameValid,
  isEmailValid,
  isPasswordValid,
  isUserRegistrationValid,
};
export { UserRegistration };
