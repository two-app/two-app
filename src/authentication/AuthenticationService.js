// @flow
import Gateway from "../http/Gateway";
import {AxiosError, AxiosResponse} from "axios";
import {UserRegistration} from "./register_workflow/UserRegistrationModel";
import {UnconnectedUser} from "./UserModel";
import {Tokens} from "./AuthenticationModel";

class RegisterUserResponse {
    user: UnconnectedUser;
    tokens: Tokens;

    constructor(user: UnconnectedUser, tokens: Tokens) {
        this.user = user;
        this.tokens = tokens;
    }
}


const registerUser = (userRegistration: UserRegistration): Promise<RegisterUserResponse> =>
    Gateway.post("/self", userRegistration)
        .then((r: AxiosResponse) => {
            const accessToken = r.data['accessToken'];
            const refreshToken = r.data['refreshToken'];
            return new RegisterUserResponse({
                user: UnconnectedUser.fromAccessToken(accessToken),
                tokens: new Tokens(accessToken, refreshToken)
            });
        }).catch((e: AxiosError) => {
            throw new Error(e.response.data['message'].toString());
        }
    );

export default {registerUser};
export {RegisterUserResponse};