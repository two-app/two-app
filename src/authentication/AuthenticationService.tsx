import Gateway from '../http/Gateway';
import {AxiosError, AxiosResponse} from 'axios';
import UserRegistrationModel, {UserRegistration} from './register_workflow/UserRegistrationModel';
import {UnconnectedUser, unconnectedUserFromAccessToken, User, userFromAccessToken, detectUserFromAccessToken} from './UserModel';
import {Tokens} from './AuthenticationModel';

export type UserResponse = {
    user: UnconnectedUser | User,
    tokens: Tokens
}

export type LoginCredentials = {
    email: string,
    rawPassword: string
}

export const areCredentialsValid = ({ email, rawPassword }: LoginCredentials) =>
    UserRegistrationModel.isEmailValid(email) && rawPassword.length > 3;

const login = (loginCredentials: LoginCredentials): Promise<UserResponse> => Gateway.post("/login", loginCredentials)
    .then((r: AxiosResponse<Tokens>): UserResponse => ({
        user: detectUserFromAccessToken(r.data.accessToken),
        tokens: {accessToken: r.data.accessToken, refreshToken: r.data.refreshToken}
    })).catch((e: AxiosError) => {
        throw new Error(e.response?.data['reason'].toString());
    });

const registerUser = (userRegistration: UserRegistration): Promise<UserResponse> => Gateway.post('/self', userRegistration)
    .then((r: AxiosResponse<Tokens>): UserResponse => ({
        user: unconnectedUserFromAccessToken(r.data.accessToken),
        tokens: {accessToken: r.data.accessToken, refreshToken: r.data.refreshToken}
    })).catch((e: AxiosError) => {
        throw new Error(e.response?.data['reason'].toString());
    });

const connectToPartner = (connectCode: String): Promise<UserResponse> => Gateway.post(`/partner/${connectCode}`)
    .then((r: AxiosResponse<Tokens>): UserResponse => ({
        user: userFromAccessToken(r.data.accessToken),
        tokens: {accessToken: r.data.accessToken, refreshToken: r.data.refreshToken}
    })).catch((e: AxiosError) => {
        // @ts-ignore
        throw new Error(e.response.data['reason'].toString());
    });

export default {login, registerUser, connectToPartner};