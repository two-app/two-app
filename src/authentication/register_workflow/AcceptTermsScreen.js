// @flow

import React, {useState} from "react";
import {ActivityIndicator, Dimensions, StyleSheet, Text, View} from "react-native";
import AcceptBox from "./AcceptSwitch";
import LogoHeader from "../LogoHeader";
import SubmitButton from "../../forms/SubmitButton";
import {WrapperContainer} from "../../views/View";
import Colors from "../../Colors";
import {UserRegistration} from "./UserRegistrationModel";
import {NavigationActions, StackActions} from "react-navigation";
import {connect} from "react-redux";
import {storeUser} from "../UserReducer";
import AuthenticationService from "../AuthenticationService";

const AcceptTermsScreen = ({navigation, storeUser}) => {
    const [userRegistration: UserRegistration, setUserRegistration] = useState(navigation.getParam("userRegistration"));
    const validAgreedState = userRegistration.acceptedTerms && userRegistration.ofAge;

    const [submitted, setSubmitted] = useState(false);
    const [registrationError, setRegistrationError] = useState(null);

    const navigateToConnectCodeScreen = () => navigation.dispatch(
        StackActions.reset({
            index: 0,
            actions: [NavigationActions.navigate({routeName: 'ConnectCodeScreen'})]
        })
    );

    if (submitted) {
        AuthenticationService.registerUser(userRegistration).then(response => {
            storeUser(response.user);
            navigateToConnectCodeScreen();
        }).catch((e: Error) => {
            setSubmitted(false);
            setRegistrationError(e.message);
        });
    }

    return (<>
        {submitted && <View style={styles.overlay}>
            <ActivityIndicator size="small" color="black" style={styles.overlayIndicator}/>
        </View>}
        <WrapperContainer>
            <LogoHeader heading="Terms & Conditions"/>
            <AcceptBox onEmit={acceptedTerms => setUserRegistration({...userRegistration, acceptedTerms})}
                       id="terms"
                       required>
                I agree to the <Text style={{fontWeight: "bold"}}>Terms & Conditions</Text> and
                <Text style={{fontWeight: "bold"}}> Privacy Policy.</Text>
            </AcceptBox>
            <AcceptBox onEmit={ofAge => setUserRegistration({...userRegistration, ofAge})}
                       id="age"
                       required>
                I am over the age of 16.
            </AcceptBox>
            <AcceptBox onEmit={receivesEmails => setUserRegistration({...userRegistration, receivesEmails})}>
                I agree to occasionally receive emails from Two.
            </AcceptBox>

            <SubmitButton onSubmit={() => setSubmitted(true)} text="Accept" disabled={!validAgreedState}/>
            {registrationError && <Text style={styles.error} id="error-message">{registrationError}</Text>}
        </WrapperContainer>
    </>);
};

AcceptTermsScreen.navigationOptions = {
    title: 'Terms & Conditions',
    header: null
};

const styles = StyleSheet.create({
    overlay: {
        flex: 1,
        position: "absolute",
        opacity: 0.5,
        backgroundColor: 'white',
        width: Dimensions.get("window").width,
        height: Dimensions.get("window").height,
        zIndex: 2,
        alignItems: "center",
        justifyContent: "center"
    },
    overlayIndicator: {
        zIndex: 3
    },
    error: {
        color: Colors.DARK_SALMON
    }
});

export default connect(null, {storeUser})(AcceptTermsScreen);
export {AcceptTermsScreen};