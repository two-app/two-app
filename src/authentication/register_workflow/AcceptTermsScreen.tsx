import React, {useState} from 'react';
import {StyleSheet, Text, View, Linking} from 'react-native';
import type {NavigationProp, RouteProp} from '@react-navigation/native';
import {useNavigation, CommonActions, useRoute} from '@react-navigation/native';
import Config from 'react-native-config';

import LogoHeader from '../LogoHeader';
import SubmitButton from '../../forms/SubmitButton';
import {ScrollContainer} from '../../views/View';
import Colors from '../../Colors';
import AuthenticationService from '../AuthenticationService';
import type {RootStackParamList} from '../../../Router';
import type {ErrorResponse} from '../../http/Response';

import type {UserRegistration} from './UserRegistrationModel';
import AcceptBox from './AcceptSwitch';

type ScreenRouteProp = RouteProp<RootStackParamList, 'AcceptTermsScreen'>;

type ScreenNavProp = NavigationProp<RootStackParamList, 'AcceptTermsScreen'>;

const AcceptTermsScreen = () => {
  const route = useRoute<ScreenRouteProp>();
  const navigation = useNavigation<ScreenNavProp>();

  const [userRegistration, setUserRegistration] = useState<UserRegistration>(
    route.params.userRegistration,
  );
  const validAgreedState =
    userRegistration.acceptedTerms && userRegistration.ofAge;

  const [submitted, setSubmitted] = useState(false);
  const [registrationError, setRegistrationError] = useState<string | null>(
    null,
  );

  const navigateToConnectCodeScreen = () =>
    navigation.dispatch(
      CommonActions.reset({
        index: 0,
        routes: [{name: 'ConnectCodeScreen'}],
      }),
    );

  if (submitted) {
    AuthenticationService.registerUser(userRegistration)
      .then(() => navigateToConnectCodeScreen())
      .catch((e: ErrorResponse) => {
        setSubmitted(false);
        setRegistrationError(e.reason);
      });
  }

  return (
    <ScrollContainer isLoading={submitted}>
      <View>
        <LogoHeader heading="Terms & Conditions" />
        <AcceptBox
          accessibilityHint="I agree to the privacy policy."
          onEmit={acceptedTerms =>
            setUserRegistration({...userRegistration, acceptedTerms})
          }
          data-testid="terms"
          required>
          I agree to the{' '}
          <Text
            style={{fontWeight: 'bold'}}
            onPress={() => Linking.openURL(Config.PRIVACY_POLICY_URL)}>
            Privacy Policy.
          </Text>
        </AcceptBox>
        <AcceptBox
          accessibilityHint="I am over the age of 16."
          onEmit={ofAge => setUserRegistration({...userRegistration, ofAge})}
          data-testid="age"
          required>
          I am over the age of 16.
        </AcceptBox>
        <AcceptBox
          accessibilityHint="I agree to occasionally receive emails from Two."
          onEmit={receivesEmails =>
            setUserRegistration({...userRegistration, receivesEmails})
          }>
          I agree to occasionally receive emails from Two.
        </AcceptBox>

        <SubmitButton
          onSubmit={() => setSubmitted(true)}
          text="Accept"
          disabled={!validAgreedState}
          accessibilityLabel="Press to submit terms and conditions"
        />
        {registrationError && (
          <Text
            style={styles.error}
            accessibilityHint={registrationError}
            accessibilityLabel="Something went wrong with your registration.">
            {registrationError}
          </Text>
        )}
      </View>
    </ScrollContainer>
  );
};

const styles = StyleSheet.create({
  error: {
    color: Colors.DARK_SALMON,
  },
});

export {AcceptTermsScreen};
