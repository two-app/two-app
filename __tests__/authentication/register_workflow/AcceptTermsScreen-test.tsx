import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {AcceptTermsScreen} from '../../../src/authentication/register_workflow/AcceptTermsScreen';
import {UserRegistration} from '../../../src/authentication/register_workflow/UserRegistrationModel';
import AuthenticationService, {UserResponse} from '../../../src/authentication/AuthenticationService';
import {CommonActions} from '@react-navigation/native';

describe('AcceptTermsScreen', () => {
    test('should maintain snapshot', () => expect(renderer.create(<AcceptTermsScreen
        navigation={{dispatch: jest.fn()} as any}
        route={{params: {userRegistration: new UserRegistration()}} as any}
        storeUnconnectedUser={jest.fn()}
        storeTokens={jest.fn()}
    />)).toMatchSnapshot());

    let tb: AcceptTermsScreenTestBed;

    beforeEach(() => tb = new AcceptTermsScreenTestBed());

    test('submit button should be disabled with t&c checked', () => {
        tb.tickTermsAndConditions();

        expect(tb.isSubmitEnabled()).toBe(false);
    });

    test('submit button should be disabled with age checked', () => {
        tb.tickAge();

        expect(tb.isSubmitEnabled()).toBe(false);
    });

    test('should enable the submit button with t&c and age accept boxes checked', () => {
        tb.tickTermsAndConditions();
        tb.tickAge();

        expect(tb.isSubmitEnabled()).toBe(true);
    });

    describe('On Submit', () => {
        const mockRegisterResponse: UserResponse = {
            user: {uid: 24, connectCode: 'testConnectCode'},
            tokens: {accessToken: 'testAccess', refreshToken: 'testRefresh'}
        };

        beforeEach(() => {
            tb.whenRegisterUserResolve(mockRegisterResponse);
            tb.checkFieldsAndSubmitForm();
        });

        test('should register the user via auth service', () => expect(AuthenticationService.registerUser)
            .toHaveBeenCalledWith({
                ...tb.userRegistration, acceptedTerms: true, ofAge: true
            }));

        test('should store the user via redux action', done => setImmediate(() => {
            expect(tb.storeUserFn).toHaveBeenCalledWith(mockRegisterResponse.user);
            done();
        }));

        test('should store the auth tokens via redux action', done => setImmediate(() => {
            expect(tb.setTokensFn).toHaveBeenCalledWith(mockRegisterResponse.tokens);
            done();
        }));

        test('should navigate to ConnectCodeScreen if successful registration', done => setImmediate(() => {
            expect(tb.dispatchFn).toHaveBeenCalledWith(
                CommonActions.reset({
                    index: 0,
                    routes: [{name: 'ConnectCodeScreen'}]
                })
            );
            done();
        }));

        test('should display overlay with loading indicator', () => expect(
            tb.wrapper.exists('LoadingView')
        ).toBe(true));

        test('should display error if auth service throws in promise', done => {
            AuthenticationService.registerUser = jest.fn().mockRejectedValue(new Error('Some API Error Message'));

            tb.checkFieldsAndSubmitForm();

            setImmediate(() => {
                expect(tb.wrapper.find('Text[data-testid=\'error-message\']').render().text()).toEqual('Some API Error Message');
                done();
            });
        });
    });
});

class AcceptTermsScreenTestBed {
    userRegistration: UserRegistration = {
        firstName: 'Gerry',
        lastName: 'Fletcher',
        email: 'admin@two.com',
        password: 'P?4Ot2ONz:IJO&%U',
        acceptedTerms: false,
        ofAge: false,
        receivesEmails: false
    };

    navigateFn = jest.fn();
    dispatchFn = jest.fn();
    storeUserFn = jest.fn();
    setTokensFn = jest.fn();

    wrapper = shallow(<AcceptTermsScreen
        navigation={{
            navigate: this.navigateFn,
            dispatch: this.dispatchFn
        } as any}
        route={{params: {userRegistration: this.userRegistration}} as any}
        storeUnconnectedUser={this.storeUserFn} storeTokens={this.setTokensFn}/>);

    tickTermsAndConditions = () => this.wrapper.find('AcceptBox[data-testid=\'terms\']').prop<(v: boolean) => void>('onEmit')(true);
    tickAge = () => this.wrapper.find('AcceptBox[data-testid=\'age\']').prop<(v: boolean) => void>('onEmit')(true);
    isSubmitEnabled = () => !this.wrapper.find('SubmitButton').prop<boolean>('disabled');
    checkFieldsAndSubmitForm = () => {
        this.tickTermsAndConditions();
        this.tickAge();
        this.wrapper.find('SubmitButton').prop<() => void>('onSubmit')();
    };

    whenRegisterUserResolve = (response: UserResponse) => {
        AuthenticationService.registerUser = jest.fn().mockResolvedValue(response);
        return this;
    };
}