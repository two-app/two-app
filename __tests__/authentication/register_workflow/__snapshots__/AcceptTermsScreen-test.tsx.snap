// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcceptTermsScreen should maintain snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "marginLeft": "5%",
            "marginRight": "5%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EE7674",
              "fontSize": 50,
              "fontWeight": "bold",
              "paddingBottom": 20,
              "paddingTop": 20,
              "textAlign": "center",
            }
          }
        >
          two.
        </Text>
        <Text
          data-testid="heading"
          style={
            Object {
              "color": "rgb(37, 37, 37)",
              "fontSize": 25,
              "fontWeight": "bold",
            }
          }
        >
          Terms & Conditions
        </Text>
        <View
          data-testid="container"
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0.1)",
                "borderRadius": 10,
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 20,
                "padding": 20,
              },
              Object {
                "borderColor": "rgb(37, 37, 37)",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(37, 37, 37)",
                "width": "80%",
              }
            }
          >
            I agree to the 
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Terms & Conditions
            </Text>
             and
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
               Privacy Policy.
            </Text>
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "width": "20%",
              }
            }
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "height": 31,
                    "width": 51,
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "scaleX": 0.8,
                      },
                      Object {
                        "scaleY": 0.8,
                      },
                    ],
                  },
                ]
              }
              value={false}
            />
          </View>
        </View>
        <View
          data-testid="container"
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0.1)",
                "borderRadius": 10,
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 20,
                "padding": 20,
              },
              Object {
                "borderColor": "rgb(37, 37, 37)",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(37, 37, 37)",
                "width": "80%",
              }
            }
          >
            I am over the age of 16.
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "width": "20%",
              }
            }
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "height": 31,
                    "width": 51,
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "scaleX": 0.8,
                      },
                      Object {
                        "scaleY": 0.8,
                      },
                    ],
                  },
                ]
              }
              value={false}
            />
          </View>
        </View>
        <View
          data-testid="container"
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "rgba(0, 0, 0, 0.1)",
                "borderRadius": 10,
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 20,
                "padding": 20,
              },
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(37, 37, 37)",
                "width": "80%",
              }
            }
          >
            I agree to occasionally receive emails from Two.
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "width": "20%",
              }
            }
          >
            <RCTSwitch
              accessibilityRole="switch"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "height": 31,
                    "width": 51,
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "scaleX": 0.8,
                      },
                      Object {
                        "scaleY": 0.8,
                      },
                    ],
                  },
                ]
              }
              value={false}
            />
          </View>
        </View>
        <TouchableOpacity
          disabled={true}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#EE7674",
                "justifyContent": "center",
                "marginBottom": 25,
                "marginTop": 25,
                "padding": 20,
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.2)",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
            Accept
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
